donald_wade/week_04/tuesday/mta3.rb

```diff
@@ -0,0 +1,48 @@
+require "pp"
+
+subway = {
+  n: ["Times Square", "34th Street", "28th Street", "23rd Street", "Union Square", "8th Avenue"],
+
+  l: ["8th Avenue", "6th Avenue", "Union Square", "3rd Street", "1st Avenue"],
+
+  l6: ["Grand Central", "33rd Street", "28th Street", "23rd Street", "Union Square", "Astor Place"],
+}
+
+def get_to_station(on, off, myline, subway)
+  line = subway[myline]
+  # fetch is a useful method by the looks of it
+  start = line.index(on)
+  finish = line.index(off)
+  if start < finish
+    route = line[start..finish]
+  elsif start > finish
+    route = line[finish..start].reverse
+  end
+  return route
+end
+
+def get_to_with_change(on, lineon, off, lineoff, subway)
+  route = [get_to_station(on, "Union Square", lineon, subway), get_to_station("Union Square", off, lineoff, subway)]
```

Comment:
you can do as so `common_station = subway[:n] & subway[:l] & subway[:l6]` to get "Union Square" out and reuse it in your code, otherwise "Union Square" becomes hardcoded value and same time becomes kind of "magic number".